src/jv_aux.c

```diff
@@ -38,29 +47,32 @@ static int parse_slice(jv j, jv slice, int* pstart, int* pend) {
       jv_get_kind(end_jv) != JV_KIND_NUMBER) {
     jv_free(start_jv);
     jv_free(end_jv);
-    return 0;
-  } else {
-    double dstart = jv_number_value(start_jv);
-    double dend = jv_number_value(end_jv);
-    jv_free(start_jv);
-    jv_free(end_jv);
-    if (dstart < 0) dstart += len;
-    if (dend < 0) dend += len;
-    if (dstart < 0) dstart = 0;
-    if (dstart > len) dstart = len;
-
-    int start = (int)dstart;
-    int end = (dend > len) ? len : (int)dend;
-    // Ends are exclusive but e.g. 1 < 1.5 so :1.5 should be :2 not :1
-    if(end < dend) end += 1;
-
-    if (end > len) end = len;
-    if (end < start) end = start;
-    assert(0 <= start && start <= end && end <= len);
-    *pstart = start;
-    *pend = end;
-    return 1;
-  }
+    return jv_invalid_with_msg(jv_string("Array/string slice indices must be integers"));
+  }
+  double dstart = jv_number_value(start_jv);
+  double dend = jv_number_value(end_jv);
+  jv_free(start_jv);
+  jv_free(end_jv);
+  if (dstart < 0) dstart += len;
+  if (dend < 0) dend += len;
+  if (dstart < 0) dstart = 0;
+  if (dstart > len) dstart = len;
+  if (dstart < INT_MIN) dstart = INT_MIN;
+  if (dstart > INT_MAX) dstart = INT_MAX;
+  if (dend   < INT_MIN) dend   = INT_MIN;
+  if (dend   > INT_MAX) dend   = INT_MAX;
```

Comment:
Is it ok to mix a double and INT_MIN/MAX? guaranteed to fit without losing precision?

Reply:
Good question.  If `sizeof(int)` is `8` then... no.  I'll fix it.

Reply:
I've pushed a fix for this.